BREAKING_CHANGES.md

```diff
@@ -32,12 +32,21 @@
 *   The `BranchPrefixToTrim` configuration property has been removed. `RegularExpression` is now used to capture named groups instead.
     *   Default `RegularExpression` for feature branches is changed from `^features?[/-]` to `^features?[/-](?<BranchName>.+)` to support using `{BranchName}` out-of-the-box
     *   Default `RegularExpression` for unknown branches is changed from `.*` to `(?<BranchName>.*)` to support using `{BranchName}` out-of-the-box
+*   The `Mainline` mode and the related implementation has been removed completely. The new `TrunkBased` version strategy should be used instead.
 *   The branch related property `is-mainline` in the configuration system has been renamed to `is-main-branch`
 *   The versioning mode has been renamed to deployment mode and consists of following values:
     *   ManualDeployment (previously ContinuousDelivery)
     *   ContinuousDelivery (previously ContinuousDeployment)
     *   ContinuousDeployment (new)
-	
+*   On the configuration root level a new property with name `version-strategy` has been introduced with following values:
+    *   ConfigNext
```

Comment:
What's the rationale behind the name `ConfigNext`?

Reply:
Good question: I have taken the name of the stragies and removed the trailing VersionStrategy postfix.
- ConfigNextVersionVersionStrategy
- MergeMessageVersionStrategy
- TaggedCommitVersionStrategy
- TrackReleaseBranchesVersionStrategy
- TrunkBasedVersionStrategy
- VersionInBranchNameVersionStrategy

Reply:
But what is `ConfigNext`? The name should give a clear indication of what it does. To me, `ConfigNext` means nothing. 🤷🏼

Reply:
That means you are saing in other words: In git version the strategy classes have not a clear name to get an indecation of for what it is good for!? Oka. It's the core domain and the most important code if you ask me. Thus it makes sense to refactor it.

I can tell you what ConfigNextVersionVersionStrategy is doing. I takes the next-version configuration property if existent and returns it is as next-version.

How would you name it?

Reply:
Aha, I see. I'm not sure that `next-version` should have its own strategy because it's a configuration parameter and not the Git repository that affects the version number. It would be a bit strange to have `next-version` configured, but not have the `ConfigNext` strategy configured, leading to `next-version` being ignored. I think the existence of `next-version` should imply that it should be considered for version calculation.

Reply:
You mean I should delete the ConfigNextVersionVersionStrategy class and place the logic somewhere else?

Reply:
I think so. There's already a lot of confusion about how `next-version` works, because it doesn't interoperate with the other strategies the way people expect. I think we need to integrate `next-version` differently somehow, and by doing that, fix the many issues people have with it. See #303 for how it was first implemented, then the following issues caused by confusion about how it works: #937, #1474, #1996, #2441, #2661, and #2670.

If we are unable to fix any of those issues in v6, I think we should just remove the property altogether (but not in this PR).

Reply:
In my opinion the concept is quite clear and works very well now also with the TrunkBased workflow. Previously we have trouble in the Mainline mode but this problem doesn't remain. Thus I don't see any reason why we should remove the feature. Just my view: The next-version only pumps to a higher version if the next calculated version is lower.

But I agree it's not part of this PR. ;)